src/libaktualizr/package_manager/dockerappmanager.cc

```diff
@@ -152,7 +152,15 @@ data::InstallationResult DockerAppManager::install(const Uptane::Target &target)
     return dapp.render(ss.str(), true) && dapp.start();
   };
   if (!iterate_apps(target, cb)) {
-    return data::InstallationResult(data::ResultCode::Numeric::kInstallFailed, "Could not render docker app");
+    res = data::InstallationResult(data::ResultCode::Numeric::kInstallFailed, "Could not render docker app");
+  }
+  if( config.docker_prune_enabled) {
+    LOG_INFO << "Pruning old docker artifacts";
+    // Utils::shell which isn't interactive, we'll use std::system so that
+    // stdout/stderr is streamed while docker sets things up.
+    if (std::system("docker system prune -a -f --volumes") != 0) {
```

Comment:
This will prune all unused images, and I wonder if this is what we want to do here (guess it should be fine for now). There is a --filter option that we can use to prune only the ones we want.

Also, --volumes scares me as it will also delete data that the user might still care. Maybe just prune the images instead?

Reply:
Our current systemd job is calling prune with 'system prune --force' only.

Reply:
I was on the fence with --volumes also. One thing I considered was making the config option be "docker_prune_cmd". Then you could get more exotic. I was just a little worried that might get (ab)used incorrectly. However, this might be a way to give power users yet another hook into controlling how updates are done. What's your opinion?